src/main/java/seedu/address/model/person/Item.java

```diff
@@ -0,0 +1,299 @@
+    public Item changeTags(HashSet<Tag> newTags) {
+        return new ItemBuilder().setItemDescription(itemDescription)
+                .setTask(task)
+                .setEvent(event)
+                .setReminder(reminder)
+                .setTags(newTags)
+                .build();
+    }
+
+    /**
+     * Returns true if both items have the same task referenced.
+     * This defines a weaker notion of equality between two items.
+     */
+    public boolean hasSameTask(Item otherItem) {
+        return getTask().equals(otherItem.getTask());
+    }
+
+    /**
+     * Returns true if both items have the same Event referenced.
+     * This defines a weaker notion of equality between two items.
+     */
+    public boolean hasSameEvent(Item otherItem) {
+        return getEvent().equals(otherItem.getEvent());
+    }
+
+    /**
+     * Returns true if both items have the same Reminder referenced.
+     * This defines a weaker notion of equality between two items.
+     */
+    public boolean hasSameReminder(Item otherItem) {
+        return getReminder().equals(otherItem.getReminder());
+    }
+
+    /**
+     * Returns true if both items have the same identity and data fields.
+     * This defines a stronger notion of equality between two items.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof Item)) {
+            return false;
+        }
+
+        Item otherItem = (Item) other;
+        return otherItem.getTask().equals(getTask())
+                && otherItem.getReminder().equals(getReminder())
+                && otherItem.getEvent().equals(getEvent())
+                && otherItem.getItemDescription().equals(getItemDescription())
+                && otherItem.getTags().equals(getTags());
+    }
+
+    @Override
+    public int hashCode() {
+        // use this method for custom fields hashing instead of implementing your own
+        return Objects.hash(task, event, reminder, itemDescription, tags);
+    }
+
+    @Override
+    public String toString() {
+        final StringBuilder builder = new StringBuilder();
+        builder.append("Description: ")
+                .append(itemDescription.toString())
+                .append(" Task: ")
+                .append(getTask().toString())
+                .append(" Event: ")
+                .append(getEvent().toString())
+                .append(" Reminder: ")
+                .append(getReminder().toString())
+                .append(" Tags: ")
+                .append(getTags().toString());
+        getTags().forEach(builder::append);
+        return builder.toString();
+    }
+
+    /**
+     * Builder class for Item.
+     */
+    public static class ItemBuilder {
+
+        // Identity fields
+        private Task task;
+        private Event event;
+        private Reminder reminder;
+
+        // Data fields
+        private ItemDescription itemDescription;
```

Comment:
Should all of these fields be initialized to null even before the itemBuilder is run?